crates/ra_ide/src/completion/complete_qualified_path.rs

```diff
@@ -91,8 +93,21 @@ pub(super) fn complete_qualified_path(acc: &mut Completions, ctx: &CompletionCon
                 }
             }
         }
+        Some(res @ PathResolution::TypeParam(_)) | Some(res @ PathResolution::SelfType(_)) => {
+            if let Some(typens) = res.in_type_ns() {
+                associated_items(ctx.db, typens, |_, _, assoc_id| {
+                    match assoc_id {
+                        AssocItemId::ConstId(c) => acc.add_const(ctx, c.into()),
+                        AssocItemId::FunctionId(f) => acc.add_function(ctx, f.into(), None),
```

Comment:
I don't think we should try to handle associated consts or functions with this, that's bound to get edge cases wrong. Those should be done through the normal `Type` methods as above.